app/controllers/predicates/AuthPredicate.scala

```diff
@@ -48,11 +48,13 @@ class AuthPredicate @Inject()(enrolmentsAuthService: EnrolmentsAuthService,
   override def invokeBlock[A](request: Request[A], block: User[A] => Future[Result]): Future[Result] = {
 
     implicit val req: Request[A] = request
+    val agentAccessAllowed: Boolean = !appConfig.agentBlockedUrls.contains(req.uri)
```

Comment:
Does request uri include query parameters, and if so could somebody use that to get around this check?

Reply:
Implementation has been changed and this issue is now avoided.